src/ripple/app/ledger/LedgerEntrySet.cpp

```diff
@@ -1406,6 +1422,40 @@ TER LedgerEntrySet::trustDelete (
     return terResult;
 }
 
+void LedgerEntrySet::enableDeferredCredits (bool enable)
+{
+    if (!enable)
+    {
+        mDeferredCredits.reset ();
+        return;
+    }
+
+    if (!mDeferredCredits)
+        mDeferredCredits.emplace ();
+}
```

Comment:
Just a thought here: Since you're going through a lot of effort to make sure that you don't nest deferring credits (ie. `ScopedDeferCredits`), would it be worth asserting at the top of this function that you're definitely flipping state, 
```assert(enable == !mDeferredCredits);```
or that if you're enabling, you're not already enabled?
```assert(!enable || !mDeferredCredits);```